realm/src/androidTest/java/io/realm/RealmResultsTest.java

```diff
@@ -231,6 +322,26 @@ public void testSumGivesCorrectValue() {
         assertEquals((TEST_DATA_SIZE - 1) * TEST_DATA_SIZE / 2, sum.intValue());
     }
 
+    // Test sum on nullable rows with all null values
+    public void testSumGivesCorrectValueForAllNullRows() {
+        TestHelper.populateAllNullRowsForNumericTesting(testRealm);
+
+        RealmResults<NullTypes> resultList = testRealm.where(NullTypes.class).findAll();
+        assertEquals(0, resultList.sum(NullTypes.FIELD_INTEGER_NULL).intValue());
+        assertEquals(0f, resultList.sum(NullTypes.FIELD_FLOAT_NULL).floatValue());
+        assertEquals(0d, resultList.sum(NullTypes.FIELD_DOUBLE_NULL).doubleValue());
```

Comment:
when asserting double you may want to use a `delta` parameter for the assert precision